lightning/src/offers/async_receive_offer_cache.rs

```diff
@@ -268,8 +269,11 @@ impl AsyncReceiveOfferCache {
 			self.reset_offer_paths_request_attempts()
 		}
 
-		self.needs_new_offer_idx(duration_since_epoch).is_some()
-			&& self.offer_paths_request_attempts < MAX_UPDATE_ATTEMPTS
+		if self.offer_paths_request_attempts >= MAX_UPDATE_ATTEMPTS {
+			return None;
+		}
+
+		self.needs_new_offer_idx(duration_since_epoch).and_then(|idx| idx.try_into().ok())
```

Comment:
Maybe panic instead or returning `None` on overflow?

Reply:
I just mean that this problem is now silently happening without any indication. I think also something worth avoiding :)